app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,22 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      if status == 'publish'
+        success_ids << id if can?(:update, media_object)
```

Comment:
what happens if you cannot update media object and you get to here (and same thing for unpublish)?

This logic will fall all the way down to line 151 `MediaObject.update_status_bulk success_ids, current_user.user_key, params`

With no flashes or success_ids/errors set?

(or maybe im confused and cancan will raise an exception here and will get caught somewhere and displayed an authorized flash message or something?)